src/server/scripts/Commands/cs_ahbot.cpp

```diff
@@ -73,102 +78,58 @@ class ahbot_commandscript : public CommandScript
         return commandTable;
     }
 
-    static bool HandleAHBotItemsAmountCommand(ChatHandler* handler, char const* args)
+    static bool HandleAHBotItemsAmountCommand(ChatHandler* handler, std::array<uint32, MAX_AUCTION_QUALITY> items)
     {
-        uint32 qVals[MAX_AUCTION_QUALITY];
-        char* arg = strtok((char*)args, " ");
-        for (int i = 0; i < MAX_AUCTION_QUALITY; ++i)
-        {
-            if (!arg)
-                return false;
-            qVals[i] = atoi(arg);
-            arg = strtok(nullptr, " ");
-        }
+        sAuctionBot->SetItemsAmount(items);
 
-        sAuctionBot->SetItemsAmount(qVals);
-
-        for (int i = 0; i < MAX_AUCTION_QUALITY; ++i)
-            handler->PSendSysMessage(LANG_AHBOT_ITEMS_AMOUNT, handler->GetTrinityString(ahbotQualityIds[i]), sAuctionBotConfig->GetConfigItemQualityAmount(AuctionQuality(i)));
+        for (AuctionQuality quality : EnumUtils::Iterate<AuctionQuality>())
+            handler->PSendSysMessage(LANG_AHBOT_ITEMS_AMOUNT, handler->GetTrinityString(ahbotQualityLangIds.at(quality)), sAuctionBotConfig->GetConfigItemQualityAmount(quality));
 
         return true;
     }
 
     template <AuctionQuality Q>
-    static bool HandleAHBotItemsAmountQualityCommand(ChatHandler* handler, char const* args)
+    static bool HandleAHBotItemsAmountQualityCommand(ChatHandler* handler, uint32 amount)
     {
-        char* arg = strtok((char*)args, " ");
-        if (!arg)
-            return false;
-        uint32 qualityVal = atoi(arg);
-
-        sAuctionBot->SetItemsAmountForQuality(Q, qualityVal);
-        handler->PSendSysMessage(LANG_AHBOT_ITEMS_AMOUNT, handler->GetTrinityString(ahbotQualityIds[Q]),
+        sAuctionBot->SetItemsAmountForQuality(Q, amount);
+        handler->PSendSysMessage(LANG_AHBOT_ITEMS_AMOUNT, handler->GetTrinityString(ahbotQualityLangIds.at(Q)),
             sAuctionBotConfig->GetConfigItemQualityAmount(Q));
 
         return true;
     }
 
-    static bool HandleAHBotItemsRatioCommand(ChatHandler* handler, char const* args)
+    static bool HandleAHBotItemsRatioCommand(ChatHandler* handler, uint32 alliance, uint32 horde, uint32 neutral)
     {
-        uint32 rVal[MAX_AUCTION_QUALITY];
-        char* arg = strtok((char*)args, " ");
-        for (int i = 0; i < MAX_AUCTION_QUALITY; ++i)
-        {
-            if (!arg)
-                return false;
-            rVal[i] = atoi(arg);
-            arg = strtok(nullptr, " ");
-        }
+        sAuctionBot->SetItemsRatio(alliance, horde, neutral);
```

Comment:
Are you sure this is correct? Why did the original iterate to `MAX_AUCTION_QUALITY`?

Reply:
Yes. The original code was just bogus. 

The values end up in
```c++
void AuctionBotSeller::SetItemsRatio(uint32 al, uint32 ho, uint32 ne)
{
    sAuctionBotConfig->SetConfig(CONFIG_AHBOT_ALLIANCE_ITEM_AMOUNT_RATIO, std::max(al, 100000u));
    sAuctionBotConfig->SetConfig(CONFIG_AHBOT_HORDE_ITEM_AMOUNT_RATIO, std::max(ho, 100000u));
    sAuctionBotConfig->SetConfig(CONFIG_AHBOT_NEUTRAL_ITEM_AMOUNT_RATIO, std::max(ne, 100000u));

    for (int i = 0; i < MAX_AUCTION_HOUSE_TYPE; ++i)
        LoadItemsQuantity(_houseConfig[i]);
}
```